### PR TITLE
[Flex] Flex flow support sample file

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
flex.yaml
```
entry: simple_callable_class:MyFlow
sample: ./inputs.json
```

```
# use sample file in flex.flow
pf.flow.test(flow="path/to/flex_flow")

# override sample
pf.flow.test(flow="path/to/flex_flow", inputs="path/to/sample.json")

pf.flow.test(flow="path/to/flex_flow", inputs={"input_1": "val_1"})

```

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-core/promptflow/core/_utils.py [hunk 1]
```diff
@@ -179,3 +181,20 @@ def get_workspace_from_resource_id(resource_id: str, credential, pkg_name: Optio
...
     )
     return ml_client.workspaces.get(name=workspace_name)
+
+
+def load_inputs_from_sample(sample: Union[dict, str, PathLike]):
+    if not sample:
+        return {}
+    elif isinstance(sample, dict):
+        return sample
+    elif isinstance(sample, (str, Path)) and Path(sample).suffix.lower() == ".json":
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/core/_utils.py:191]: note we need to add some reference doc for this feature

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/tests/test_configs/eager_flows/basic_callable_class_with_sample_file/inputs.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"func_input": "mock_input"}
```
Comment (crazygao) [inline src/promptflow/tests/test_configs/eager_flows/basic_callable_class_with_sample_file/inputs.json:1]: add some negative example samples
func input is a number or json?
or input is not named as func_input

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow-devkit/promptflow/_sdk/entities/_flows/flex.py [hunk 3]
```diff
@@ -45,6 +45,10 @@ def language(self) -> str:
     def additional_includes(self) -> list:
         return self._data.get("additional_includes", [])
 
+    @property
```
Comment (D-W-) [inline src/promptflow-devkit/promptflow/_sdk/entities/_flows/flex.py:48]: Also add in flex flow's schema?

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow-devkit/promptflow/_sdk/entities/_flows/flex.py (hunk 3 above)
Comment (lalala123123) [reply src/promptflow-devkit/promptflow/_sdk/entities/_flows/flex.py:48]: image
it has been added in #2622

**Review by D-W- — APPROVED** [review]

**Pull request merged**